modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/BinaryProcessor.cs

```diff
@@ -165,16 +165,17 @@ public BinaryType RegisterEnum(string typeName, IEnumerable<KeyValuePair<string,
         /// Gets the type name by id.
         /// </summary>
         /// <param name="id">The identifier.</param>
+        /// <param name="registerSameJavaType">True if should register type both for dotnet and java platforms.</param>	
         /// <returns>Type or null.</returns>
-        public string GetTypeName(int id)
+        public string GetTypeName(int id, bool registerSameJavaType)
         {
             try
             {
                 return GetTypeName(id, DotNetPlatformId);
             }
             catch (BinaryObjectException)
             {
-                if (!Marshaller.RegisterSameJavaType.Value)
+                if (!registerSameJavaType)
```

Comment:
Let's avoid using exceptions for control flow. At least we can push the flag to `PlatformBinaryProcessor.OP_GET_TYPE` handler, which already has an exception handler.

Reply:
Fixed.